### PR TITLE
Add escaping features in unhandled endpoints

### PR DESCRIPTION
**Pull request by nofun97**
Changes proposed in this pull request:
- Add escaping features in unhandled endpoints in call statements and collector statements (related to #658 )

Checklist:
- [x] Added related tests
- [x] Made corresponding changes to the documentation

@anz-bank/sysl-developers

### CODE REVIEW

File: pkg/parse/listener_impl.go [hunk 1]
```diff
@@ -3150,3 +3147,11 @@ func (s *TreeShapeListener) currentType() *sysl.Type {
 func (s *TreeShapeListener) setCurrentType(type1 *sysl.Type) {
 	s.typemap[s.fieldname[len(s.fieldname)-1]] = type1
 }
+
+func mustUnescape(endpoint string) string {
```
Comment (joshcarp) [inline pkg/parse/listener_impl.go]: MustEscape + MustUnescape

File: pkg/parse/listener_impl.go [hunk 2]
```diff
@@ -1688,15 +1684,16 @@ func (s *TreeShapeListener) EnterCollector_call_stmt(ctx *parser.Collector_call_
 		Stmt: &sysl.Statement_Call{
 			Call: &sysl.Call{
 				Target:   appName,
-				Endpoint: strings.TrimSpace(ctx.Target_endpoint().GetText()),
+				Endpoint: strings.TrimSpace(mustUnescape(ctx.Target_endpoint().GetText())),
```
Comment (joshcarp) [inline pkg/parse/listener_impl.go]: maybe should be within mustUnescape

File: pkg/parse/Parser_test.go [hunk 3]
```diff
@@ -239,6 +239,7 @@ func parseAndCompare(
 	if err = pbutil.FTextPB(&actual, module); err != nil {
 		return false, err
 	}
+	fmt.Println(actual.String())
```
Comment (joshcarp) [inline pkg/parse/Parser_test.go]: println

File: pkg/importer/utils.go [hunk 4]
```diff
@@ -77,6 +77,9 @@ func getSyslSafeEndpoint(endpoint string) string {
 		"2F": "/",
 		"7B": "{",
 		"7D": "}",
+		"3D": "=",
+		"3F": "?",
+		"26": "&",
 	}
 	endpoint = url.PathEscape(endpoint)
 	endpoint = strings.ReplaceAll(endpoint, ".", "%"+"2E")
```
Comment (joshcarp) [inline pkg/importer/utils.go]: get rid of string +

File: pkg/parse/listener_impl.go (hunk 1 above)
Comment (nofun97) [reply pkg/parse/listener_impl.go]: there's `url.PathEscape` that doesn't return error

**Comment by joshcarp** [comment]
Nice

**Pull request merged**